src/createRoom.ts

```diff
@@ -119,7 +119,9 @@ export default async function createRoom(client: MatrixClient, opts: IOpts): Pro
     const createOpts: ICreateRoomOpts = opts.createOpts || {};
     createOpts.preset = createOpts.preset || defaultPreset;
     createOpts.visibility = createOpts.visibility || Visibility.Private;
-    if (opts.dmUserId && createOpts.invite === undefined) {
+
+    // We allow UX of DMing ourselves as a form of creating a personal room but the server doesn't like this so strip it
```

Comment:
Nit: I think this comment could be more clear to say the server doesn't like the invite in particular